src/main/java/com/direwolf20/buildinggadgets/common/util/ref/Reference.java

```diff
@@ -3,6 +3,19 @@
 import com.direwolf20.buildinggadgets.api.APIReference;
 import net.minecraft.util.ResourceLocation;
 
+/**
+ * @implNote @michaelhillcox: So much is wrong here I can event start to fix it but for future me
+ *           the MODID is being applied when almost every single instance uses a class annotation
+ *           that defines the same MODID.
+ *
+ *           Either remove it here or remove it there. Either would fix this.
+ *
+ *           Why in the literal hell are we pulling our MODID reference from the API. That is
+ *           completely backwards.
```

Comment:
I don't see why that's backwards... The API cannot and should not access the main mods code, therefore this is the correct order to do it. If you want to, you can go through and replace all usages with APIReference.MODID, but I just didn't feel like it at the time.